python/tank/log.py

```diff
@@ -557,9 +557,34 @@ def initialize_base_file_handler(self, log_name):
         :returns: The name of the previous log_name that is being switched away from,
                   None if no base logger was previously active.
         """
+        # avoid cyclic references
+        from .util import filesystem
+
+        return self.initialize_base_file_handler_from_path(
+            os.path.join(
+                self.log_folder,
+                "%s.log" % filesystem.create_valid_filename(log_name)
+            )
+        )
+
+    def initialize_base_file_handler_from_path(self, log_file):
+        """
+        Create a file handler and attach it to the sgtk base logger.
+
+        This method is there for legacy Toolkit applications and shouldn't be used. Use
+        ``initialize_base_file_handler`` instead.
+
+        :param log_file: Path of the file to write the logs to.
+
+        :returns: The name of the previous log_name that is being switched away from,
```

Comment:
The current pattern where we use it is this:

```
prev_name = uninitialize_base_file_handler()
do_stuff_not_logged
initialize_base_file_handler(prev_name)
```

This is why I think it makes sense to maintain a symmetry. Plus it's tricky to change the API now after the release :) Hmm, but i see your point about not being able to return the value now....